apps/dashboard/app/models/project.rb

```diff
@@ -27,33 +27,86 @@ def find(project_path)
     def dataroot
       Rails.logger.debug("project path is: #{OodAppkit.dataroot.join('projects')}")
 
-      OodAppkit.dataroot.join('projects').tap do |p|
+      OodAppkit.dataroot.join('projects').tap do |path|
         p.mkpath unless p.exist?
       rescue StandardError => e
         Pathname.new('')
       end
     end
   end
 
+  validates :dir, presence: true
+  validates :dir, format: {
+
+    with: /[\w-]+\z/,
+    message: 'Name may only contain letters, digits, dashes, and underscores'
+  }
+
   attr_reader :dir
+  delegate :icon, :name, :description, to: :manifest
+
+  def initialize(attributes = {})
+    @dir            = attributes.fetch(:dir, nil).to_s
+  end
 
-  def initialize(args = {})
-    @dir = args.fetch(:dir, nil).to_s
+  def save
+    begin
+      Rails.logger.debug("Writing manifest to: #{manifest_path}")
+      write_manifest
+    rescue StandardError => error
+      Rails.logger.debug("Writting manifest failed: #{error}")
+    rescue Errno::ENOENT => error
+      Rails.logger.debug("manifest.yml file not found: #{error}")
+    end
+    true
```

Comment:
Removed the `make_manifest` method since it wasn't needed and tried to introduce some basic cases of error handling. Are there other cases I'm missing here you can think of?

Reply:
Todo: fix the spelling error in above.